job-runner/runner/runner.go

```diff
@@ -269,7 +283,24 @@ func (r *runner) Runtime() float64 {
 	return time.Now().Sub(r.startTime).Seconds()
 }
 
-func (r *runner) Status() string {
-	r.logger.Infof("getting job status")
-	return r.realJob.Status() // this is a blocking operation that should return quickly
+func (r *runner) Status() Status {
+	// Get real-time status before locking in case it's slow
+	status := r.realJob.Status()
+
+	r.Lock()
+	defer r.Unlock()
+
+	// Indicate in real-time status if job is sleep, i.e. not truly running
+	sleeping := r.retrySleep
+	if sleeping {
```

Comment:
What's the reason behind not just checking `if r.retrySleep`?

Reply:
Copy-paste error. Fixed.